uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeaveRequest.php

```diff
@@ -897,23 +897,48 @@ public static function findOverlappingLeaveRequests($leaveRequestParams, $leaveR
     $halfDayAMID = $leaveRequestDayTypes['half_day_am'];
     $halfDayPMID = $leaveRequestDayTypes['half_day_pm'];
     $isTOIL = $leaveRequestParams['request_type'] === LeaveRequest::REQUEST_TYPE_TOIL;
+    $isCalculationUnitInHoursOrTOIL = $isCalculationUnitInHours || $isTOIL;
 
     foreach ($absenceTypes as $absenceType) {
+      $absenceTypesIDs[] = $absenceType->id;
       if ($leaveRequestAbsenceType->calculation_unit === $absenceType->calculation_unit) {
         $absenceTypesIDsWithSameUnit[] = $absenceType->id;
       } else {
         $absenceTypesIDsWithDifferentUnit[] = $absenceType->id;
       }
     }
 
-    $noOverlappingCondition =
-      $isCalculationUnitInHours || $isTOIL
-      ?
-        "(lr.from_date < '{$toDate}' AND lr.to_date > '{$fromDate}'" . (!$isTOIL ? " AND lr.type_id IN (" . join(',', $absenceTypesIDsWithSameUnit) . ")" : "") . ")" .
-        (count($absenceTypesIDsWithDifferentUnit) && !$isTOIL ? " OR ((lrd.date BETWEEN %3 AND %4) AND lr.type_id IN (" . join(',', $absenceTypesIDsWithDifferentUnit) . "))" : "")
-      : "lrd.date BETWEEN %3 AND %4 AND lr.type_id IN (" . join(',', array_merge($absenceTypesIDsWithSameUnit, $absenceTypesIDsWithDifferentUnit)) . ")" .
-        ($fromDateType !== null && $fromDateType === $halfDayPMID ? " AND (DATE_FORMAT(lr.to_date, '%y-%m-%d') != DATE_FORMAT('{$fromDate}', '%y-%m-%d') OR lr.to_date_type != {$halfDayAMID})" : "") .
-        ($toDateType !== null && $toDateType === $halfDayAMID ? " AND (DATE_FORMAT(lr.from_date, '%y-%m-%d') != DATE_FORMAT('{$toDate}', '%y-%m-%d') OR lr.from_date_type != {$halfDayPMID})" : "");
+    $dateTimeSelector = $isCalculationUnitInHoursOrTOIL
+      ? "lr.from_date < '{$toDate}' AND lr.to_date > '{$fromDate}'"
```

Comment:
Just thinking about the date formats here, When this function is used in the Leave Request BAO validation it will pass in dates already processed by CRM_Utils_Date::processDate and when the function is called by another function it may accept date in another format. For consistency and so that we would be comparing dates in same format. I would suggest we convert the from date and to date to dateTime objects.
That way here, we can use the sql query parameters %3 and %4 for fromDate and toDate respectively, 
```php
      3 => [CRM_Utils_Date::processDate($fromDate, null, false, 'Y-m-d'), 'String'],
       4 => [CRM_Utils_Date::processDate($toDate, null, false, 'Y-m-d'), 'String'],
``` 
remember to change this to something like:
```php
      3 => [$fromDate->format('Y-m-d H:i:s', 'String'],
       4 => [$toDate->format('Y-m-d H:i:s', 'String'],
```
In the considerHalfDayRequests variable you can maintain the condition and instead use 
`!= {$fromDate->format('Y-m-d')})`.
This would ensure the datetime format is kept consistent throughout the function and under control.

Reply:
It will not work for Days, it needs to be "lr.from_date <= '{$toDate}' AND lr.to_date >= '{$fromDate}'". I went a bit another way than you suggested.